cpp_utils/test/unittest/math/random/randomTest.cpp

```diff
@@ -0,0 +1,206 @@
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <array>
+#include <algorithm>
+#include <cstdlib>
+
+#include <gtest_aux.hpp>
+#include <gtest/gtest.h>
+
+#include <cpp_utils/math/random/RandomManager.hpp>
+
+using namespace eprosima::utils;
+
+namespace test {
+
+constexpr const unsigned int TEST_ITERATIONS = 100;
+
+} /* namespace test */
+
+/**
+ * Test the generation of std random numbers from a default seed.
+ *
+ * This will use the \c std::rand method to know the N first values that are expected.
+ *
+ * STEPS:
+ * - calculates the N first numbers from rand()
+ * - create N Random Manager and generate N values that are equal the rand() N ones
+ *
+ * NOTE: this test could also be done by seeding rand to 1.
+ */
+TEST(randomTest, get_random_sequence_number)
+{
+    // calculates the N first numbers from rand()
+    RandomManager initial_manager;
+    std::array<RandomNumberType, test::TEST_ITERATIONS> rand_numbers;
+    for (unsigned int i=0; i<test::TEST_ITERATIONS; ++i)
+    {
+        rand_numbers[i] = initial_manager.rand();
+    }
+
+    // create N Random Manager and generate N values that are equal the N first ones
+    for (unsigned int i=0; i<test::TEST_ITERATIONS; ++i)
+    {
+        RandomManager manager;
+        for (unsigned int j=0; j<test::TEST_ITERATIONS; ++j)
+        {
+            ASSERT_EQ(manager.rand(), rand_numbers[j]) << "Manager " << i << " in iteration " << j;
+        }
+    }
+}
+
+/**
+ * Test that the N first pure random numbers are not included in the first N values generated
+ * by std::rand .
+ *
+ * STEPS:
+ * - calculates the N first numbers from rand()
+ * - create Random Manager and generate N values with method rand<true> that should not be in N first generated
+ * - create Random Manager and generate N values with method pure_rand that should not be in N first generated
+ */
+TEST(randomTest, get_pure_random_number)
+{
+    // calculates the N first numbers from rand()
+    std::array<RandomNumberType, test::TEST_ITERATIONS> rand_numbers;
+    for (unsigned int i=0; i<test::TEST_ITERATIONS; ++i)
+    {
+        rand_numbers[i] = rand();
+    }
+
+    // create Random Manager and generate N values with method rand<true> that should not be in N first generated
+    {
+        RandomManager manager;
+        for (unsigned int i=0; i<test::TEST_ITERATIONS; ++i)
+        {
+            // Generate new value
+            auto new_value = manager.rand<true>();
+
+            // Check that new value is not in array
+            ASSERT_EQ(
+                std::find(std::begin(rand_numbers), std::end(rand_numbers), new_value),
+                std::end(rand_numbers));
```

Comment:
I am not sure this test makes sense (other than to increase coverage), you are testing for repetition, which is a probabilistic event depending on precision (given by `unsigned int`) and `test::TEST_ITERATIONS`.

Reply:
Indeed, it tests that pure random numbers are generated, and that it is not a subset or a different loop for other seed.
Obviously it could fail randomly.

Reply:
I don't know... maybe it would be better to assert generated sequences do not coincide (which still could occur but is way less probable than just checking no number gets repeated). Anyways, not important. But if we add the xfail label then I think every test with a `ASSERT_NE` should have it (all but `get_random_sequence_number`).

Reply:
I added deterministic tests and marked these as xfail